parley/src/layout/line_break.rs

```diff
@@ -56,8 +57,38 @@ struct PrevBoundaryState {
     state: LineState,
 }
 
-#[derive(Clone, Default)]
-struct BreakerState {
+/// Reason that the line breaker has yielded control flow
+pub enum YieldData {
+    /// Control flow was yielded because a line break was encountered
+    LineBreak(LineBreakData),
+    /// Control flow was yielded because content on the line caused the line to exceed the max height
+    MaxHeightExceeded(MaxHeightBreakData),
+    /// Control flow was yielded because an inline box with kind `InlineBoxKind::CustomOutOfFlow`
+    /// was encountered.
+    InlineBoxBreak(BoxBreakData),
+}
+
+pub struct LineBreakData {
+    pub reason: BreakReason,
+    pub advance: f32,
+    pub line_height: f32,
+}
+
+pub struct MaxHeightBreakData {
+    pub advance: f32,
+    pub line_height: f32,
+}
+
+pub struct BoxBreakData {
+    /// The user-supplied ID for the inline box
+    pub inline_box_id: u64,
+    // The index of the inline box within the inline_boxes `Vec`
+    pub inline_box_index: usize,
+    pub advance: f32,
+}
+
+#[derive(Clone)]
+pub struct BreakerState {
```

Comment:
Since this is now public, could you please add some short documentation about how it might be used?